test/cpp/jit/test_gpu.cpp

```diff
@@ -1067,13 +1150,691 @@ void testGPU_FusionLoopUnroll() {
+
+    torch::jit::fuser::cuda::compileKernel(fusion, &prog);
+    torch::jit::fuser::cuda::runTestKernel(&prog, inputs, outputs);
+
+    ref_output = at::lerp(input1, input2, input3);
+
+    TORCH_CHECK(output.allclose(ref_output),
+                "\nOp Type: -- ", "lerp",
+                " -- had a mismatch.\n",
+                "IN1 : ", input1, "\n",
+                "IN2 : ", input2, "\n",
+                "IN3 : ", input3, "\n",
+                "JIT: ", output, "\n",
+                "REF: ", ref_output, "\n");
+  }
+  {
+    Fusion fusion;
+    FusionGuard fg(&fusion);
+
+    std::vector<IterDomain*> dom;
+    for (int i = 0; i < 2; i++)
+      dom.push_back(new IterDomain(new Int(0), new Int()));
+
+    Float*       f1 = new Float(1.f);
+    TensorView* tv0 = new TensorView(new TensorDomain(dom), DataType::Float);
+    TensorView* tv1 = new TensorView(new TensorDomain(dom), DataType::Float);
+    TensorView* tv2 = new TensorView(new TensorDomain(dom), DataType::Float);
+    TensorView* tv3 = static_cast<TensorView*>(addcmul(tv0, tv1, tv2, f1));
+
+    fusion.addInput(tv0);
+    fusion.addInput(tv1);
+    fusion.addInput(tv2);
+    fusion.addOutput(tv3);
+    tv0->computeAt(tv3, -1);
+    tv1->computeAt(tv3, -1);
+    tv2->computeAt(tv3, -1);
+
+    tv3->axis(0)->parallelize(ParallelType::BIDx);
+    tv3->axis(-1)->parallelize(ParallelType::TIDx);
+
+    torch::jit::fuser::cuda::CudaKernel prog;
+    prog.device_ = 0;
+    prog.grid(1);
+    prog.block(4);
+
+    auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+    at::Tensor input1     = at::rand({1,4}, options);
+    at::Tensor input2     = at::rand({1,4}, options);
+    at::Tensor input3     = at::rand({1,4}, options);
+    at::Tensor output     = at::empty_like(input1);
+    at::Tensor ref_output = at::empty_like(input1);
+
+    std::vector<at::Tensor> inputs{{input1, input2, input3}};
+    std::vector<at::Tensor> outputs{{output}};
+
+    torch::jit::fuser::cuda::compileKernel(fusion, &prog);
+    torch::jit::fuser::cuda::runTestKernel(&prog, inputs, outputs);
+
+    ref_output = at::addcmul(input1, input2, input3, 1.f);
+
+    TORCH_CHECK(output.allclose(ref_output),
+                "\nOp Type: -- ", "addcmul",
+                " -- had a mismatch.\n",
+                "IN1 : ", input1, "\n",
+                "IN2 : ", input2, "\n",
+                "IN3 : ", input3, "\n",
+                "JIT: ", output, "\n",
+                "REF: ", ref_output, "\n");
+  }
+}
+
+void testGPU_FusionTernaryOps() {
+  {
+    Fusion fusion;
+    FusionGuard fg(&fusion);
+
+    std::vector<IterDomain*> dom;
+    for (int i = 0; i < 2; i++)
+      dom.push_back(new IterDomain(new Int(0), new Int()));
+
+    Float*       f0 = new Float(0.f);
+    Float*       f1 = new Float(1.f);
+    TensorView* tv0 = new TensorView(new TensorDomain(dom), DataType::Float);
+    TensorView* tv1 = static_cast<TensorView*>(clamp(tv0, f0, f1));
```

Comment:
yep, we probably should update the range.

IIUC, this is the unit test where we intended to keep thing as simple as possible. Hence we don't want to add other binary ops into the equation. cc'ing @kevinstephano in case I got anything wrong.